### PR TITLE
doc/install: update about ninja

### PR DESCRIPTION
**Pull request by varshar16**
'ninja' replaced 'make' in this PR[1]. This patch updates the doc about it.

[1] #39826

Signed-off-by: Varsha Rao <varao@redhat.com>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/install/build-ceph.rst [hunk 1]
```diff
@@ -38,17 +38,17 @@ repository and execute the following::
...
 .. note:: By default do_cmake.sh will build a debug version of ceph that may
    perform up to 5 times slower with certain workloads. Pass 
    '-DCMAKE_BUILD_TYPE=RelWithDebInfo' to do_cmake.sh if you would like to
    build a release version of the ceph executables instead.
 
-.. topic:: Hyperthreading
-
-	You can use ``make -j`` to execute multiple jobs depending upon your system. For 
-	example, ``make -j4`` for a dual core processor may build faster.
+   If unspecified ninja determines the number of jobs based on the number of CPU
```
Comment (tchaikov) [inline doc/install/build-ceph.rst:48]: i'd avoid replicating the document.

File: README.md [hunk 2]
```diff
@@ -60,12 +60,12 @@ Build instructions:
 (Note: do_cmake.sh now defaults to creating a debug build of ceph that can
 be up to 5x slower with some workloads. Please pass 
 "-DCMAKE_BUILD_TYPE=RelWithDebInfo" to do_cmake.sh to create a non-debug
-release.)
+release.
```
Comment (tchaikov) [inline README.md:63]: i'd drop the "Note: ". from the text.

File: doc/install/build-ceph.rst (hunk 1 above)
Comment (varshar16) [reply doc/install/build-ceph.rst]: I have removed it.

File: README.md (hunk 2 above)
Comment (varshar16) [reply README.md:63]: Done

File: doc/install/build-ceph.rst [hunk 3]
```diff
@@ -38,17 +38,7 @@ repository and execute the following::
...
-.. note:: By default do_cmake.sh will build a debug version of ceph that may
-   perform up to 5 times slower with certain workloads. Pass 
-   '-DCMAKE_BUILD_TYPE=RelWithDebInfo' to do_cmake.sh if you would like to
-   build a release version of the ceph executables instead.
-
-.. topic:: Hyperthreading
-
-	You can use ``make -j`` to execute multiple jobs depending upon your system. For 
-	example, ``make -j4`` for a dual core processor may build faster.
+    ninja
```
Comment (rishabh-d-dave) [inline doc/install/build-ceph.rst:41]: Is this note outdated? If not, it's good to have it.

**Review by rishabh-d-dave — APPROVED** [review]

File: doc/install/build-ceph.rst [hunk 4]
```diff
@@ -38,17 +38,7 @@ repository and execute the following::
     cd ceph
     ./do_cmake.sh
     cd build
-    make
-
-.. note:: By default do_cmake.sh will build a debug version of ceph that may
-   perform up to 5 times slower with certain workloads. Pass 
-   '-DCMAKE_BUILD_TYPE=RelWithDebInfo' to do_cmake.sh if you would like to
-   build a release version of the ceph executables instead.
```
Comment (tchaikov) [inline doc/install/build-ceph.rst:46]: this change is not related to the ninja change. why would you want to remove it? i cannot find any explanation in the commit message.

Comment (varshar16) [reply doc/install/build-ceph.rst:46]: I removed it as `README.md` has the same note. I have created a new commit for this change and added a link to the `README` doc .

File: doc/install/build-ceph.rst (hunk 3 above)
Comment (varshar16) [reply doc/install/build-ceph.rst]: Yes, hyperthreading part of the note is outdated. I have added a note to refer `README`.

**Review by tchaikov — APPROVED** [review]

**Comment by varshar16** [comment]
jenkins test make check
